Allow prod/dev/test environments

kenfehling opened the pull request:
Allows to use local MongoDB for testing and development and the remote MongoDB for production. Also eliminates needing to set COOKIE_SECRET in testing and development.

pgeiss commented:
Perhaps edit the readme to reflect the new changes? The process to start up is different with this patch, I'd imagine. Also, the URI I've posted isn't for prod, it's for dev (Peter wanted a unified DB for dev so we didn't have to keep making dummy data) but I'm not opposed to just setting the NODE_ENV to prod to access it (I'm not sure if that variable changes anything else in the Keystone system, however).

kingofirony commented on server.js:
```diff
@@ -7,26 +7,39 @@ require('dotenv').load();
+}
+
+if (env !== 'production') {
+	keystone.set('cookie secret', 'whatever, it is not production');
```
lol.

kingofirony commented on server.js:
```diff
@@ -7,26 +7,39 @@ require('dotenv').load();
+	}
+	else if (env === 'test') {
+		return 'mongodb://localhost:27017/wusb_test';
+	}
```
what does this return if env doesn't equal one of the three - is it possible that it could? maybe have a case just incase that happens

kenfehling commented:
@pgeiss We could seed the database using updates (similar to how the default user is created), so then when you run the project in development it has whatever test data we want in there. Actually we should also be doing something similar in `0.0.1-admins.js` to check if the `env` is production so it doesn't create the default user.

kenfehling commented:
@kingofirony OK, good idea. When it's not set to 1 of the 3 it'll now throw an error.

pgeiss commented:
@kenfehling Feel free to edit 0.0.1-admins.js to ensure env isn't prod. You can get it using process.env.NODE_ENV ([source](http://stackoverflow.com/questions/9441287/how-to-find-out-the-current-node-env-the-express-app-is-running-under))

kenfehling commented:
OK, I've edited `0.0.1-admins.js`, added `0.0.2-posts.js` for seeding the DB in development mode, and updated the README to show the setup. I think that's all the remaining issues with this one.
